test/e2e/operator_test.go

```diff
@@ -191,17 +198,24 @@ func TestExternalDNSRecordLifecycle(t *testing.T) {
 		}
 
 		// get the IPs of the loadbalancer
-		lbHostname := service.Status.LoadBalancer.Ingress[0].Hostname
-		// use built in Go resolver instead of the platform's one
-		ips, err := customResolver("").LookupIP(context.TODO(), "ip", lbHostname)
-		if err != nil {
-			t.Logf("waiting for loadbalancer IP for %s", lbHostname)
-			// if the hostname cannot be resolved currently then retry later
+		if service.Status.LoadBalancer.Ingress[0].Hostname != "" {
+			lbHostname := service.Status.LoadBalancer.Ingress[0].Hostname
+			// use built in Go resolver instead of the platform's one
+			ips, err := customResolver("").LookupIP(context.TODO(), "ip", lbHostname)
+			if err != nil {
+				t.Logf("waiting for loadbalancer IP for %s", lbHostname)
+				// if the hostname cannot be resolved currently then retry later
+				return false, nil
+			}
+			for _, ip := range ips {
+				serviceIPs[ip.String()] = struct{}{}
+			}
+		} else if service.Status.LoadBalancer.Ingress[0].IP != "" {
```

Comment:
I think it makes more sense to check the presence of IP first and only if it's not there check the hostname.

Reply:
actually did same, but changed :), will do the revert.